### PR TITLE
Journalist can create article feature

### PR DESCRIPTION
**Pull request by Felix-Willborg**
### User Story
```
As a journalist,
In order to write articles
I would like to be able to have an article creation form
```
## Changes proposed in this pull request:
*  Created admin login functionality and creating new articles.  
## What I have learned working on this feature:
* to upload images in an "article" and to stub out file upload in cypress
## Packages/Gems added
- Axios & Cypress file upload package
## Screenshots (Client)
Screenshot 2020-08-11 at 22 53 30
Screenshot 2020-08-11 at 22 53 13

### CODE REVIEW

**Review by Alex-Bellamy — COMMENTED** [review]
looks great, nice one people

**Review by Cma-B — APPROVED** [review]
nice!

**Review by Odeane — APPROVED** [review]
Good to go 👌🏽

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  beforeEach(() => {
+    cy.server();
+    cy.visit("/")
+    cy.route({
+      method: "POST",
+      url: "http://localhost:3000/api/v1/articles",
+      response: '{"message": "Your article was successfully created"}',
+    });
+  });
+  it("journalist can create an article successfully", () => {
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: You need to add a sad path

File: cypress/fixtures/registration_response.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "data": {
```
Comment (oliverochman) [inline cypress/fixtures/registration_response.json]: why is this added if you don't have an authentication flow?

File: src/App.jsx [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+import { Header, Icon } from "semantic-ui-react"
 
-const App = () => {
-  return (
-    <div>
-    </div>
-  );
+class App extends Component {
+  state = {
+    authenticated: false,
```
Comment (oliverochman) [inline src/App.jsx]: you don't use this state, is it necessary?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    cy.get("button").contains("Create Article").click();
+    cy.get("#article-form").within(() => {
+      cy.get("#title").type("Title");
+      cy.get("#lead").type("Lead");
+      cy.get("#category").click()
+      cy.get("div[role='option']").contains("Lifestyle").click()
+      cy.get("#content").type("This is content");
+      cy.file_upload("img.jpeg", "#image-upload", "image/jpeg");
+      cy.get("button").contains("Save Article").click();
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: whats the assertion here?

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    });
+  });
+});
+
+describe("Journalist can't create article", () => {
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: this should not be a describe block

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  });
+  it("journalist can create an article successfully", () => {
+    cy.get("button").contains("Create Article").click();
+    cy.get("#article-form").within(() => {
+      cy.get("#lead").type("Lead");
+      cy.get("#category").click()
+      cy.get("div[role='option']").contains("Lifestyle").click()
+      cy.get("#content").type("This is content");
+      cy.file_upload("img.jpeg", "#image-upload", "image/jpeg");
+      cy.get("button").contains("Save Article").click();
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: whats the assertion here?

**Review by oliverochman — CHANGES_REQUESTED** [review]
